consensus/polybft/burnt_fees_manager.go

```diff
@@ -0,0 +1,140 @@
+type burntFeesManager struct {
+	// key is the identity of the node submitting a checkpoint
+	key ethgo.Key
+
+	// txRelayer is the abstraction on the child chain interaction logic.
+	txRelayer txrelayer.TxRelayer
+
+	// calculateBurnContract returns burnt contract address based on the given block
+	calculateBurnContract func(block uint64) (types.Address, error)
+
+	// forks is needed to check if london hardfork is enabled for the given block
+	forks *chain.Forks
+
+	// checkpointsOffset represents offset between withdrawal blocks (applicable only for non-epoch ending blocks)
+	withdrawalOffset uint64
+
+	// lastWithdrawalBlock represents the last block on which a withdrawal transaction was sent
+	lastWithdrawalBlock uint64
+
+	// logger is the default logger
+	logger hclog.Logger
+}
+
+// newBurntFeesManager is the constructor of burntFeesManager
+func newBurntFeesManager(
+	key ethgo.Key,
+	txRelayer txrelayer.TxRelayer,
+	withdrawalOffset uint64,
+	calculateBurnContract func(block uint64) (types.Address, error),
+	forks *chain.Forks,
+	logger hclog.Logger,
+) *burntFeesManager {
+	return &burntFeesManager{
+		key:                   key,
+		txRelayer:             txRelayer,
+		calculateBurnContract: calculateBurnContract,
+		forks:                 forks,
+		withdrawalOffset:      withdrawalOffset,
+		logger:                logger,
+	}
+}
+
+func (m *burntFeesManager) PostBlock(req *PostBlockRequest) error {
+	latestHeader := req.FullBlock.Block.Header
+
+	// Check if london hardfork is enabled
+	if !m.forks.IsLondon(latestHeader.Number) {
+		return nil
+	}
+
+	// Check if withdrawal is needed by the current miner
+	if !m.isWithdrawalBlock(latestHeader.Number, req.IsEpochEndingBlock) ||
+		!bytes.Equal(m.key.Address().Bytes(), latestHeader.Miner) {
+		return nil
+	}
+
+	m.logger.Debug("burnt fees withdrawal invoked...",
+		"withdrawal block", latestHeader.Number)
+
+	// Get burn contract address based on the given header
+	burnContractAddr, err := m.calculateBurnContract(latestHeader.Number)
+	if err != nil {
+		return err
+	}
+
+	burnContractAddrPtr := ethgo.Address(burnContractAddr)
+	txn := &ethgo.Transaction{
+		To:   &burnContractAddrPtr,
+		From: m.key.Address(),
+	}
+
+	// Encode transaction input
+	input, err := contractsapi.EIP1559Burn.Abi.GetMethod("withdraw").Encode([]interface{}{})
+	if err != nil {
+		return err
+	}
+
+	txn.Input = input
+
+	// Send withdrawal transaction
+	receipt, err := m.txRelayer.SendTransaction(txn, m.key)
+	if err != nil {
+		return err
+	}
+
+	if receipt.Status == uint64(types.ReceiptFailed) {
+		return fmt.Errorf("burnt fees withdrawal transaction failed for block %d", latestHeader.Number)
+	}
+
+	// Update last withdrawal block
+	m.lastWithdrawalBlock = req.FullBlock.Block.Number()
+
+	// update burnt fees withdrawal block number metrics
+	metrics.SetGauge([]string{"bridge", "checkpoint_block_number"}, float32(latestHeader.Number))
```

Comment:
Did you want to name the metric something else? You probably forgot to rename it.

Reply:
Good point, done